subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/TestSummaryPluginBuildStep.kt

```diff
@@ -0,0 +1,65 @@
+package com.avito.ci.steps
+
+import com.android.build.gradle.internal.tasks.factory.dependsOn
+import com.avito.android.stats.statsdConfig
+import com.avito.impact.configuration.internalModule
+import com.avito.instrumentation.extractReportCoordinates
+import com.avito.instrumentation.instrumentationTask
+import com.avito.logger.GradleLoggerFactory
+import com.avito.report.model.ReportCoordinates
+import com.avito.test.summary.TestSummaryExtension
+import com.avito.test.summary.TestSummaryFactory
+import com.avito.test.summary.testSummaryPluginId
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.tasks.TaskContainer
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.kotlin.dsl.getByType
+
+abstract class TestSummaryPluginBuildStep(context: String, name: String) :
+    BuildStep(context, name),
+    ImpactAnalysisAwareBuildStep by ImpactAnalysisAwareBuildStep.Impl() {
+
+    var configuration: String = ""
+
+    protected abstract val stepName: String
+
+    protected abstract fun getOrCreateTask(
+        rootTasksContainer: TaskContainer,
+        extension: TestSummaryExtension,
+        reportCoordinates: ReportCoordinates,
+        testSummaryFactory: TestSummaryFactory,
+    ): TaskProvider<out Task>
+
+    override fun registerTask(project: Project, rootTask: TaskProvider<out Task>) {
+        require(project.rootProject.pluginManager.hasPlugin(testSummaryPluginId)) {
+            "$stepName step can't be initialized without $testSummaryPluginId plugin applied to root project"
+        }
+
+        require(configuration.isNotBlank()) {
+            "$stepName step can't be initialized without provided instrumentation configuration"
+        }
+
+        if (useImpactAnalysis && !project.internalModule.isModified()) return
```

Comment:
out of scope, but can be checked in execution phase

Reply:
it's common pattern in steps, i don't like it too